src/library_memfs.js

```diff
@@ -364,11 +364,12 @@ mergeInto(LibraryManager.library, {
             }
           }
           allocated = true;
+          var fromHeap = (buffer.buffer == HEAP8.buffer);
```

Comment:
How about adding a helper function for this? I suspect it may be needed in more places. It may also be clearer to read.

I am thinking the code could look like this:
```
var viewGetter = FS.makeGrowAwareViewGetter(buffer);
[..]
viewGetter().set(contents, ptr);
```
And the impl would be something like
```
function makeGrowAwareViewGetter(view) {
#if ALLOW_MEMORY_GROWTH
  if (view.buffer === HEAP8.buffer) {
    return function() {
      return HEAP8; // this should check the view item type/size though
    };
  }
#endif
  return function() {
    return view;
  };
}
```
(note: all these buffers should be called views, I think, as they are typed array views, and not buffers)

Reply:
That looks a bit overarchitecting, and it will raise extra questions of what the concept of a grow-aware view getter would be. This is the only place in JS libraries where this occurs - the special case here is that the function passes a heap buffer as local variable, but then also mallocs in between. Most code would do only one or the other, I don't think this would be that common of a scenario - if it turns out to be, then I think it might warrant factoring out commonalities, but that would be too eager to do now? (plus would increase code size)

Reply:
Fair enough, maybe it's a bit much.

Please add a comment here though, otherwise someone not aware of the growth issue might be confused.

Reply:
check